sbom-utility-scripts/scripts/add-image-reference-script/add_image_reference.py

```diff
@@ -139,16 +166,22 @@ def update_component_in_cyclonedx_sbom(sbom: dict, image: Image) -> dict:
     Returns:
         dict: Updated SBOM with the image reference added.
     """
-    # Add the image component to the components list
     image_component = {
         "type": "container",
         "name": image.name,
         "purl": image.purl(),
         "version": image.tag,
         "hashes": [{"alg": image.digest_algo_cyclonedx, "content": image.digest_hex_val}],
     }
-    sbom["components"].insert(0, image_component)
-    sbom["metadata"]["component"] = image_component
+    if image.is_builder_image:
+        # Add builder image property to image_component
+        image_component["properties"]: [BUILDER_IMAGE_PROPERTY]
```

Comment:
I caught a last minute bug :sweat:

```suggestion
        image_component["properties"] = [BUILDER_IMAGE_PROPERTY]
```

Otherwise, the property doesn't get rendered in the SBOM.

Reply:
The unit test needs to be fixed as well.

Reply:
Interesting, I made the same stupid mistake on the test and that's why the test did not catch it. Thanks, great catch @brunoapimentel !

Fixed